contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java

```diff
@@ -250,12 +250,7 @@ public void setup(OperatorContext context)
     consumer.create();
     // reset the offsets to checkpointed one
     if (consumer instanceof SimpleKafkaConsumer && !offsetStats.isEmpty()) {
-      Map<KafkaPartition, Long> currentOffsets = new HashMap<>();
-      // Increment the offsets and set it to consumer
-      for (Map.Entry<KafkaPartition, Long> e: offsetStats.entrySet()) {
-        currentOffsets.put(e.getKey(), e.getValue() + 1);
-      }
-      ((SimpleKafkaConsumer)consumer).resetOffset(currentOffsets);
+      ((SimpleKafkaConsumer)consumer).resetOffset(offsetStats);
```

Comment:
These are not mine - I ended up merging changes from upstream/master (because of conflicts) which show up in the pull request for this branch.